core/src/edu/cornell/gdiac/main/SlopeModel.java

```diff
@@ -105,13 +115,33 @@ public float getSlopeAngle() {
   }
 
   /**
-   * Returns the force applied to the player when they touch the slope, in the direction opposite of
-   * the hypotenuse
+   * Returns the "normal" force-influenced applied to the frozen player when they touch the slope. This effectively should
+   * increase the physics interaction between the player and slope.
    *
-   * @return the force applied to the player when they touch the slope, in the direction opposite of
-   * the hypotenuse
+   * @param x x component of the velocity of contact object with slope
+   * @param y y component fo the velocity of contact object with slope
+   * @return the force applied to the player when they touch the slope.
    */
-  public float getSlopeFrozenImpulse() {
-    return frozenImpulse;
+  public Vector2 getSlopeFrozenNormalImpulse(float x, float y) {
+    // Note: Frozen impulse effectively "magnifies" the effect of gravity when in contact with slope
+
+    boolean isLeft = slopeAngle >= 0 && slopeAngle <= Math.PI;
+    if (isLeft) {
+      // Slope is pointing down left
+      v2Cache.set(
+        (float) -Math.cos(slopeAngle),
+        (float) -Math.sin(slopeAngle)
+      );
+    } else {
+      // Slope is pointing down right
+      v2Cache.set((float) Math.cos(slopeAngle),
+        (float) Math.sin(slopeAngle));
+    }
+
+    float maxVel = 10;
```

Comment:
I feel like this should be tunable depending on the slope, but also 10 does seem like more than enough.